src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/CredentialsMaskingBatchTest.java

```diff
@@ -0,0 +1,300 @@
+    private static final String ESCAPE = "^<^>^(^)^^^&^|";
+
+    // ALL_ASCII - [ < > " ' ^ $ | % ] 
+    private static final String NON_DANGEROUS = "!#$*+,-./ 0123456789:;=? @ABCDEFGHIJKLMNO PQRSTUVWXYZ[\\]_ `abcdefghijklmno pqrstuvwxyz{}~";
+
+    // <>"'^$|
+    private static final String NON_DANGEROUS_IN_DOUBLE = "abc<def>$ghi|jkl";
+
+    private static final String ALL_ASCII = "!\"#$%&'()*+,-./ 0123456789:;<=>? @ABCDEFGHIJKLMNO PQRSTUVWXYZ[\\]^_ `abcdefghijklmno pqrstuvwxyz{|}~";
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private WorkflowJob project;
+
+    private String credentialPlainText;
+    private String credentialId;
+
+    @Before
+    public void assumeWindowsForBatch() throws Exception {
+        assumeTrue(Functions.isWindows());
+    }
+
+    private void registerCredentials(String password) throws IOException {
+        this.credentialPlainText = password;
+        this.credentialId = UUID.randomUUID().toString();
+        StringCredentials credentials = new StringCredentialsImpl(CredentialsScope.GLOBAL, credentialId, null, Secret.fromString(password));
+        CredentialsProvider.lookupStores(j.jenkins).iterator().next().addCredentials(Domain.global(), credentials);
+    }
+
+    private WorkflowRun runProject() throws Exception {
+        return j.assertBuildStatusSuccess(project.scheduleBuild2(0));
+    }
+
+    @Test
+    public void simple_noQuote() throws Exception {
+        registerCredentials(SIMPLE);
+        assertDirectNoPlainTextButStars(runDirectNoQuote());
+    }
+
+    @Test
+    public void simple_singleQuote() throws Exception {
+        registerCredentials(SIMPLE);
+        assertDirectNoPlainTextButStars(runDirectSingleQuote());
+    }
+
+    @Test
+    public void simple_doubleQuote() throws Exception {
+        registerCredentials(SIMPLE);
+        assertDirectNoPlainTextButStars(runDirectDoubleQuote());
+    }
+
+    @Test
+    public void simple_delayed() throws Exception {
+        registerCredentials(SIMPLE);
+        assertDelayedNoPlainTextButStars(runDelayedAllQuotes());
+    }
+
+    @Test
+    public void nonDangerous_noQuote() throws Exception {
+        registerCredentials(NON_DANGEROUS);
+        assertDirectNoPlainTextButStars(runDirectNoQuote());
+    }
+
+    @Test
+    public void nonDangerous_singleQuote() throws Exception {
+        registerCredentials(NON_DANGEROUS);
+        assertDirectNoPlainTextButStars(runDirectSingleQuote());
+    }
+
+    @Test
+    public void nonDangerous_doubleQuote() throws Exception {
+        registerCredentials(NON_DANGEROUS);
+        assertDirectNoPlainTextButStars(runDirectDoubleQuote());
+    }
+
+    @Test
+    public void nonDangerous_delayed() throws Exception {
+        registerCredentials(NON_DANGEROUS);
+        assertDelayedNoPlainTextButStars(runDelayedAllQuotes());
+    }
+
+    @Test
+    @Ignore("Cannot support the dangerous characters in direct expressions")
```

Comment:
What's the value of having these ignored tests?

Reply:
@Wadeck had initially authored them, probably to show all the possibilities?

Reply:
Yeah and to show the "impossibility" to support the dangerous stuff.

Reply:
I don't think the ignored tests do a good job of showing that. It's basically introducing dead code that we never expect to fix. I'd prefer a different approach. One possibility would be to remove the `@Ignore`s, implement the test to expect the known behavior, and then leave some comments or naming to indicate that while this isn't the desired behavior it is expected. At least this way the tests do something, affirming the behavior continues.

Reply:
I tried updating them to check for a failed build. Let's see how that works out.

Reply:
@jvz  just in case, I sent you the patch to apply that modification (https://cloudbees.slack.com/archives/DBZE45TCM/p1557400247010700)